File: app/src/main/java/com/example/madarsofttask/data/repositories/UserRepository.java
package com.example.madarsofttask.data.repositories;

import androidx.lifecycle.MutableLiveData;
import com.example.madarsofttask.data.models.User;
import java.util.List;
import static com.example.madarsofttask.application.BaseApplication.userDatabase;

public class UserRepository {
    private static UserRepository instance;

    public static UserRepository getInstance(){
       if(instance == null){
           instance = new UserRepository();
       }
       return instance;
    }

    public MutableLiveData<List<User>> getUsers(){
        List<User> users = userDatabase.myDAO().retrieveUsers();
        MutableLiveData<List<User>> usersData = new MutableLiveData<>();
        usersData.setValue(users);
        return usersData;
    }

    public void saveUser(User user) {
        userDatabase.myDAO().addUser(user);
    }
}

File: app/src/main/java/com/example/madarsofttask/data/repositories/UserRepository.kt
package com.example.madarsofttask.data.repositories

import androidx.lifecycle.MutableLiveData
import com.example.madarsofttask.application.BaseApplication
import com.example.madarsofttask.data.models.User

class UserRepository {
    val users: MutableLiveData<List<User?>?>
        get() {
            val users = BaseApplication.userDatabase?.myDAO()?.retrieveUsers()
            val usersData = MutableLiveData<List<User?>?>()
            usersData.value = users
            return usersData
        }

    fun saveUser(user: User?) {
        BaseApplication.userDatabase?.myDAO()?.addUser(user)
    }

    companion object {
        @JvmStatic
        var instance: UserRepository? = null
            get() {
                if (field == null) {
                    field = UserRepository()
                }
                return field
            }
            private set
    }
}